src/main/java/picard/sam/BamErrorMetric/BaseErrorCalculation.java

```diff
@@ -0,0 +1,266 @@
+                    .stream()
+                    .filter(putative -> areReadsMates(record, putative.getRecord()))
+                    .findFirst()
+                    .orElse(null);
+
+            // we are only interested in bases for which the mate read also has a base over the same locus
+            if (mate == null) return;
+            // both bases need to be called for this error calculation
+
+            final byte mateBase = mate.getReadBase();
+            if (SequenceUtil.isNoCall(readBase)) return;
+            if (SequenceUtil.isNoCall(mateBase)) return;
+
+            nTotalBasesWithOverlappingReads++;
+
+            // Only bases that disagree with the reference are counted as errors.
+            if (SequenceUtil.basesEqual(readBase, locusAndRef.getReferenceBase())) return;
+
+            final boolean agreesWithMate = SequenceUtil.basesEqual(readBase, mateBase);
+            final boolean mateAgreesWithRef = SequenceUtil.basesEqual(mateBase, locusAndRef.getReferenceBase());
+
+            if (agreesWithMate) {
+                nBothDisagreeWithReference++;
+            } else if (mateAgreesWithRef) {
+                nDisagreeWithRefAndMate++;
+            } else {
+                nThreeWaysDisagreement++;
+            }
+        }
+
+        /**
+         * The suffix that pertains to the implementation of aggregation
+         **/
+        @Override
+        public String getSuffix() {
+            return "overlapping_error";
+        }
+
+        /**
+         * Returns the metric generated by the observed loci
+         **/
+        @Override
+        public ErrorMetrics.OverlappingErrorMetric getMetric() {
+            return new ErrorMetrics.OverlappingErrorMetric("", totalBases, nTotalBasesWithOverlappingReads, nDisagreeWithRefAndMate,
+                    nBothDisagreeWithReference, nThreeWaysDisagreement);
+        }
+
+        private boolean areReadsMates(final SAMRecord read1, final SAMRecord read2) {
+            // must have same name
+            if (!read1.getReadName().equals(read2.getReadName())) return false;
+            // must be paired
+            if (!read1.getReadPairedFlag()) return false;
+            // one must be first while the other is not
+            if (!read1.getFirstOfPairFlag() ^ read2.getFirstOfPairFlag()) return false;
+            // one must be second while the other is not
+            if (!read1.getSecondOfPairFlag() ^ read2.getSecondOfPairFlag()) return false;
+            // read1 must be mapped
+            if (read1.getReadUnmappedFlag()) return false;
+            // read2 must be mapped
+            if (read2.getReadUnmappedFlag()) return false;
+            // read1 must be non-secondary
+            if (read1.isSecondaryAlignment()) return false;
+            // read2 must be non-secondary
+            if (read2.isSecondaryAlignment()) return false;
+
+            return read1.getMateAlignmentStart() == read2.getAlignmentStart();
```

Comment:
I think the use of `^` here is unnecessary and makes the code harder to understand. Unless I'm not understanding it. Also I find it's easier to follow as a single `return` but maybe that's just me. e.g.
```
    static private boolean areReadsMates(final SAMRecord read1, final SAMRecord read2) {
        // must have same name
        return read1.getReadName().equals(read2.getReadName()) &&
                // must be paired
                read1.getReadPairedFlag() &&
                // one must be first while the other is not
                !read1.getFirstOfPairFlag() == read2.getFirstOfPairFlag() &&
                // one must be second while the other is not
                !read1.getSecondOfPairFlag() == read2.getSecondOfPairFlag() &&
                // read1 must be mapped
                !read1.getReadUnmappedFlag() &&
                // read2 must be mapped
                !read2.getReadUnmappedFlag() &&
                // read1 must be non-secondary
                !read1.isSecondaryAlignment() &&
                // read2 must be non-secondary
                !read2.isSecondaryAlignment() &&
                read1.getMateAlignmentStart() == read2.getAlignmentStart();
    }
```